mono/metadata/w32file.c

```diff
@@ -431,7 +431,7 @@ typedef struct {
 } IncrementalFind;
 	
 static gboolean
-incremental_find_check_match (IncrementalFind *handle, WIN32_FIND_DATA *data, MonoString **result, MonoError *error)
+incremental_find_check_match (IncrementalFind *handle, WIN32_FIND_DATA *data, MonoStringHandle *result, MonoError *error)
```

Comment:
`MonoStringHandle *` is not idiomatic.  Pass a `MonoStringHandleOut` (just a typedef for `MonoStringHandle`) and use `MONO_HANDLE_ASSIGN` to write an object into it.

Reply:
It segfaulted when I tried that, I guess because the handle is initially null? I’ll look into it more.

Reply:
Sorry about that:
1. Allocate a handle in the caller.
2. pass it to `incremental_find_check_match`
2. in `incremental_find_check_match`:
  a. `HANDLE_FUNCTION_ENTER();`
  b. `MONO_HANDLE_ASSIGN (result, mono_string_new_handle (...));`
  c. `HANDLE_FUNCTION_RETURN_VAL (whatever);`